apm-agent-core/src/main/java/co/elastic/apm/impl/transaction/Transaction.java

```diff
@@ -77,6 +84,10 @@
      * Transactions that are 'sampled' will include all available information. Transactions that are not sampled will not have 'spans' or 'context'. Defaults to true.
      */
     private boolean noop;
+    
+    public Transaction(CoreConfiguration coreConfiguration) {
```

Comment:
Remove constructor and read the span limit in `#start` via `tracer.getConfig(CoreConfiguration.class).getTransactionMaxSpans()`